gpii/node_modules/dbOperation/src/DbDataStoreUtils.js

```diff
@@ -0,0 +1,545 @@
+        error = gpii.dbOperation.composeError(gpii.dbOperation.errors.missingInput, {fieldName: "prefsSafeId"});
+        promiseTogo.reject(error);
+    } else if (!prefsSafeData) {
+        error = gpii.dbOperation.composeError(gpii.dbOperation.errors.missingDoc, {docName: gpii.dbOperation.docTypes.prefsSafe});
+        promiseTogo.reject(error);
+    } else {
+        data = {
+            type: prefsSafeData.type,
+            schemaVersion: prefsSafeData.schemaVersion,
+            prefsSafeType: prefsSafeData.prefsSafeType,
+            name: prefsSafeData.name || null,
+            password: prefsSafeData.password || null,
+            email: prefsSafeData.email || null,
+            preferences: prefsSafeData.preferences,
+            timestampCreated: prefsSafeData.timestampCreated,
+            timestampUpdated: gpii.dbOperation.getCurrentTimestamp()
+        };
+
+        promiseTogo = gpii.dbOperation.dbDataStore.updateRecord(saveDataSource, gpii.dbOperation.docTypes.prefsSafe, prefsSafeId, data);
+    }
+    return promiseTogo;
+};
+
+// General Client Functions
+// ------------------------
+
+/**
+ * Find a client by an OAuth2 client id
+ * @param data {Component} Contains both client and client credential information associated with the given oauth2 client id
+ * An input example:
+ * {
+ *     key: {String},   // access token
+ *     id: {String},    // authorization id
+ *     value: {
+ *         _id: {String},      // client id
+ *         clientCredential: {
+ *             type: {String},
+ *             schemaVersion: {String},
+ *             clientId: {String},
+ *             oauth2ClientId: {String},
+ *             oauth2ClientSecret: {String},
+ *             revoked: {Boolean},
+ *             revokedReason: {String},
+ *             timestampCreated: {Date},
+ *             timestampRevoked: {Date},
+ *             _id: {String},
+ *             _rev: {String}
+ *         }
+ *     },
+ *     doc: {
+ *         type: {String},     // client type
+ *         schemaVersion: {String},
+ *         name: {String},
+ *         computerType: {String},
+ *         timestampCreated: {Date},
+ *         timestampUpdated: {Date},
+ *         _id: {String},
+ *         _rev: {String}
+ *     }
+ * }
+ * @return {Promise} An object that carries the client and authorization information associated with the access token.
+ */
+gpii.dbOperation.dbDataStore.findClientByOauth2ClientIdPostProcess = function (data) {
+    var result;
+
+    if (data && data.doc && data.value) {
+        result = {};
+        fluid.set(result, "oauth2ClientId", data.key);
+        fluid.set(result, "client", gpii.dbOperation.dbDataStore.cleanUpDoc(data.doc));
+        fluid.set(result, "clientCredential", gpii.dbOperation.dbDataStore.cleanUpDoc(data.value.clientCredential));
+    }
+
+    return result;
+};
+
+// General Authorization Functions
+// ------------------------------------
+
+/**
+ * Find an authorization by an access token
```

Comment:
This comment does not appear to match the function. The function does some transformation on the data argument but does not query the database (there is no "Find"). And it does not return a promise.